### ISSUE TITLE
Fuzzing integration by way of OSS-Fuzz

### ISSUE THREAD

**Issue by DavidKorczynski** [ostreedev/ostree#2579]
Hi,

I was think that it would be nice to set up continuous fuzzing of ostree by way of OSS-Fuzz. In this PR: https://github.com/google/oss-fuzz/pull/7557 I have done exactly that, namely created the necessary logic from an OSS-Fuzz perspective to integrate ostree. This includes developing initial fuzzers as well as integrating into OSS-Fuzz. Down the line it is preferable to have the fuzzers upstream so we could also move all of the fuzzing infrastructure here -- I would be happy to keep them in the OSS-Fuzz repository for a few weeks while I mature the integration by expanding the fuzzers and adding more.

Essentially, OSS-Fuzz is a free service run by Google that performs continuous fuzzing of important open source projects. The only expectation of integrating into OSS-Fuzz is that bugs will be fixed. This is not a "hard" requirement in that no one enforces this and the main point is if bugs are not fixed then it is a waste of resources to run the fuzzers, which we would like to avoid.

If you would like to integrate, the only thing I need is as list of email(s) that will get access to the data produced by OSS-Fuzz, such as bug reports, coverage reports and more stats. Notice the emails affiliated with the project will be public in the OSS-Fuzz repo, as they will be part of a configuration file.

Fuzzing was briefly mentioned previously on ostree here: https://github.com/ostreedev/ostree/pull/2504#pullrequestreview-842908778

**Comment by cgwalters** [comment]
Hi David, thanks so much for this!  I wonder why the tooling can't support filing a github issue for example.  But, if it only does email right now, it's fine to start with mine I think.  Anyone else interested can comment here and add themselves.

**Comment by DavidKorczynski** [comment]
Thanks @cgwalters -- Github issue filing is just on the doorstep for OSS-Fuzz though, and the first batch of projects experimenting with this feature is ongoing (https://github.com/google/oss-fuzz/pull/7531). Am happy make sure that works as well once it has been tested through. 

Just to clarify, when you say your email you mena the email from your git commits?

**Comment by cgwalters** [comment]
> Just to clarify, when you say your email you mean the email from your git commits?

Yep!  walters@verbum.org - thanks!

**Comment by dbnicholson** [comment]
> > Just to clarify, when you say your email you mean the email from your git commits?
> 
> Yep! [walters@verbum.org](mailto:walters@verbum.org) - thanks!

I'd be interested in seeing the results. dbn@endlessos.org

**Comment by DavidKorczynski** [comment]
Thanks @cgwalters and @dbnicholson -- will let you know here once the integration is merged into OSS-Fuzz

**Comment by DavidKorczynski** [comment]
The fuzzers have now been merged https://github.com/google/oss-fuzz/pull/7557! I will do more work on it by refining and extending the fuzzers -- if there are any findings then you will receive emails with info! Am closing this issue but feel free to write here for questions/queries
